pkg/kubelet/certificate/certificate_store.go

```diff
@@ -307,3 +313,22 @@ func loadX509KeyPair(certFile, keyFile string) (*tls.Certificate, error) {
 	cert.Leaf = certs[0]
 	return &cert, nil
 }
+
+// remove obsolete certificates belong to self, but reserve current one.
+func (s *fileStore) removeObsolete(filename string) error {
+	files, err := utilfile.ReadDirNoStat(s.certDirectory)
```

Comment:
Why not just join `s.certDirectory` and `filename` to get the full path and delete it then?

Reply:
filename indicates file we can not delete, because `current` links to it.